Servers/UI/OJS.Servers.Ui/ClientApp/src/components/contests/contest-categories/ContestCategories.tsx

```diff
@@ -72,6 +82,68 @@ const ContestCategories = ({
         updateBreadcrumb(category, categoriesFlat);
     }, [ possibleFilters, categoriesFlat, onCategoryClick, updateBreadcrumb ]);
 
+    const strategyFilterGroup = useMemo(
+        () => possibleFilters.filter(({ type }) => type === FilterType.Strategy),
+        [ possibleFilters ],
+    );
+
+    const filterProps = useMemo(
+        () => [] as any,
+        [],
+    );
+
+    const addNewStrategyFilters = useCallback(
+        (id: string, node: ICategoryStrategiesTypes) => {
+            const strategyFiltersToAdd: IFilter[] = [];
+            node.allowedStrategyTypes.forEach((value) => {
+                const currAllowedStrategyId = value.id.toString();
+
+                const strategyToFind = strategyFilterGroup.find((x) => x.value === currAllowedStrategyId);
+
+                if (!isNil(strategyToFind)) {
+                    strategyFiltersToAdd.push(strategyToFind);
+                }
+            });
+
+            filterProps.push({
+                categoryId: id,
+                strategies: strategyFiltersToAdd,
+            });
+
+            const flatteredStrategyFilters = filterProps.flatMap((s: Anything) => s.strategies);
+
+            setCurrentCurrentStrategyFilters(filterProps);
+
+            setStrategyFilters(Array.from(new Set(flatteredStrategyFilters)));
```

Comment:
This is strange. Why is there a need for `Array.from(new Set(....))`? Isn't `flatteredStrategyFilters` good enough?

If the reason is __uniqueness__, better use [lodash's `uniq`](https://lodash.com/docs/4.17.15#uniq)